R/helpers_github.R

```diff
@@ -1,177 +1,181 @@
-  errors <- vapply(parsed$errors, `[[`, "message", FUN.VALUE = character(1))
-
-  structure(
-    list(
-      call = sys.call(-1),
-      message = paste0(parsed$message, " (", httr::status_code(req), ")\n",
-                       if (length(errors) > 0) {
-                         paste("* ", errors, collapse = "\n")
-                       })
-    ), class = c("condition", "error", "github_error"))
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-github_GET <- function(path, ..., pat = github_pat(),
-                       host = "https://api.github.com") {
-
-  url <- httr::parse_url(host)
-  url$path <- paste(url$path, path, sep = "/")
-  ## May remove line below at release of httr > 1.1.0
-  url$path <- gsub("^/", "", url$path)
-  ##
-  req <- httr::GET(url, github_auth(pat), ...)
-  github_response(req)
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-github_POST <- function(path, body, ..., pat = github_pat(),
-                        host = "https://api.github.com") {
-
-  url <- httr::parse_url(host)
-  url$path <- paste(url$path, path, sep = "/")
-  ## May remove line below at release of httr > 1.1.0
-  url$path <- gsub("^/", "", url$path)
-  ##
-  req <- httr::POST(url, body = body, github_auth(pat), encode = "json", ...)
-  github_response(req)
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-github_rate_limit <- function() {
-  req <- github_GET("rate_limit")
-  core <- req$resources$core
-
-  reset <- as.POSIXct(core$reset, origin = "1970-01-01")
-  cat(core$remaining, " / ", core$limit,
-      " (Reset ", strftime(reset, "%H:%M:%S"), ")\n", sep = "")
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-github_commit <- function(username, repo, ref = "master") {
-  github_GET(file.path("repos", username, repo, "commits", ref))
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-github_tag <- function(username, repo, ref = "master") {
-  github_GET(file.path("repos", username, repo, "tags", ref))
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-# Retrieve Github personal access token.
-# A github personal access token
-# Looks in env var \code{GITHUB_PAT}
-github_pat <- function(quiet = FALSE) {
-  pat <- Sys.getenv("GITHUB_PAT")
-  if (nzchar(pat)) {
-    if (!quiet) {
-      message("Using GitHub PAT from envvar GITHUB_PAT")
-    }
-    return(pat)
-  }
-  if (in_ci()) {
-    pat <- paste0("b2b7441d",
-                  "aeeb010b",
-                  "1df26f1f6",
-                  "0a7f1ed",
-                  "c485e443")
-    if (!quiet) {
-      message("Using bundled GitHub PAT. Please add your own PAT to the env var `GITHUB_PAT`")
-    }
-    return(pat)
-  }
-  return(NULL)
-}
-
-# from https://github.com/r-lib/devtools/blob/master/R/github.R
-in_ci <- function() {
-  nzchar(Sys.getenv("CI"))
-}
+# from https://github.com/hadley/devtools/blob/master/R/git.R
+uses_github <- function(path = ".") {
```

Comment:
The function names in this file are technically not correct any more and I don't know if all of these functions work interchangeably for github and gitlab. Do they, @ntrlshrp?